go/test/endtoend/vreplication/vreplication_vtctldclient_cli_test.go

```diff
@@ -184,6 +193,135 @@ func createMoveTables(t *testing.T, sourceKeyspace, targetKeyspace, workflowName
+		createFlags:  createFlags,
+	}, workflowFlavorVtctld)
+
+	ksWorkflow := fmt.Sprintf("%s.%s", keyspace, workflowName)
+	rs.Create()
+	validateReshardResponse(rs)
+	workflowResponse := getWorkflow(keyspace, workflowName)
+	reshardShowResponse := getReshardShowResponse(&rs)
+	require.EqualValues(t, reshardShowResponse, workflowResponse)
+	validateReshardWorkflow(t, workflowResponse.Workflows)
+	waitForWorkflowState(t, vc, fmt.Sprintf("%s.%s", keyspace, workflowName), binlogdatapb.VReplicationWorkflowState_Stopped.String())
+	rs.Start()
+	waitForWorkflowState(t, vc, ksWorkflow, binlogdatapb.VReplicationWorkflowState_Stopped.String())
+	for _, tab := range targetTabs {
+		alias := fmt.Sprintf("zone1-%d", tab.TabletUID)
+		query := "update _vt.vreplication set source := replace(source, 'stop_after_copy:true', 'stop_after_copy:false') where db_name = 'vt_customer' and workflow = '" + workflowName + "'"
+		output, err := vc.VtctlClient.ExecuteCommandWithOutput("ExecuteFetchAsDba", alias, query)
+		require.NoError(t, err, output)
+	}
+	rs.Start()
+	waitForWorkflowState(t, vc, fmt.Sprintf("%s.%s", keyspace, workflowName), binlogdatapb.VReplicationWorkflowState_Running.String())
+	rs.Stop()
+	waitForWorkflowState(t, vc, ksWorkflow, binlogdatapb.VReplicationWorkflowState_Stopped.String())
+	rs.Start()
+	waitForWorkflowState(t, vc, fmt.Sprintf("%s.%s", keyspace, workflowName), binlogdatapb.VReplicationWorkflowState_Running.String())
+	for _, targetTab := range targetTabs {
+		catchup(t, targetTab, workflowName, "Reshard")
+	}
+	vdiff(t, keyspace, workflowName, "zone1", false, true, nil)
+
+	rs.SwitchReadsAndWrites()
+	waitForLowLag(t, keyspace, workflowName+"_reverse")
+	vdiff(t, keyspace, workflowName+"_reverse", "zone1", true, false, nil)
+
+	rs.ReverseReadsAndWrites()
+	waitForLowLag(t, keyspace, workflowName)
+	vdiff(t, keyspace, workflowName, "zone1", false, true, nil)
+	rs.SwitchReadsAndWrites()
+	rs.Complete()
+}
+
+func getReshardShowResponse(rs *iReshard) *vtctldatapb.GetWorkflowsResponse {
+	(*rs).Show()
+	reshardOutput := (*rs).GetLastOutput()
+	var reshardResponse vtctldatapb.GetWorkflowsResponse
+	err := protojson.Unmarshal([]byte(reshardOutput), &reshardResponse)
+	require.NoError(vc.t, err)
+	reshardResponse.Workflows[0].MaxVReplicationTransactionLag = 0
+	reshardResponse.Workflows[0].MaxVReplicationLag = 0
+	return reshardResponse.CloneVT()
+}
+
+func validateReshardResponse(rs iReshard) {
+	resp := getReshardResponse(rs)
+	require.NotNil(vc.t, resp)
+	require.NotNil(vc.t, resp.ShardStreams)
+	require.Equal(vc.t, len(resp.ShardStreams), 2)
+	keyspace := "customer"
+	for _, shard := range []string{"-40", "40-80"} {
+		streams := resp.ShardStreams[fmt.Sprintf("%s/%s", keyspace, shard)]
+		require.Equal(vc.t, 1, len(streams.Streams))
+		require.Equal(vc.t, binlogdatapb.VReplicationWorkflowState_Stopped.String(), streams.Streams[0].Status)
+	}
+}
+
+func validateReshardWorkflow(t *testing.T, workflows []*vtctldatapb.Workflow) {
```

Comment:
IMO we should pass `rs *iReshard` so it's clear what reshard workflow we're validating and we can replace the string literals in here with variables from `rs`.